packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -33,6 +86,19 @@ export interface AssetOptions extends assets.CopyOptions {
    * @experimental
    */
   readonly sourceHash?: string;
+
+  /**
+   * Bundle the asset by executing a command in a Docker container.
+   * The asset path will be mounted at `/asset-input`. The Docker
+   * container is responsible for putting content at `/asset-output`.
+   * The content at `/asset-output` will be zipped and used as the
+   * final asset.
+   *
+   * @default - asset path is zipped as is
+   *
+   * @experimental
+   */
+  readonly bundling?: BundlingOptions;
```

Comment:
I don't feel comfortable with an experimental property of this impact in an API that's been stable for a while.
By stable API, I mean the `Code` class in the `aws-lambda` package.
 
The customer visibility on this is very low. There's already the default assumption that everything in the `lambda` package is stable. Further, the `Code` APIs have been around for a while now for customers to assume they are all stable by default.

Can we instead keep the asset bundling separate from the existing code APIs?
An ok next step would be to go with `Code.fromBundle()` and mark this as experimental, but still too close to the stable APIs to make assumptions about its stability.
We could go one step further out with something like -

```ts
const outputdir = AssetBundler.bundle({ ... });
Code.fromAsset(outputdir);
```

There's a bit of loss of ergonomics but there's some form of opt-in.

@eladb - thoughts?

Reply:
We have multiple experimental APIs in stable modules. That's why we have that capability. If we have a visibility problem, we should address it, but I wouldn't want to sacrifice ergonomics for that reason. One of the things we are experimenting with is ergonomics!